### ISSUE TITLE
Reading action params is not possible in matchCallback in ACF

### ISSUE THREAD

**Issue by trejder** [yiisoft/yii2#8964]
When using `matchCallback` in ACF, an attempt to read `$action->controller->actionParams` or `Yii::$app->controller->actionParams` fails, because these attributes are empty:

![asgwew4hwsr](https://cloud.githubusercontent.com/assets/2903392/8453003/71fd44d2-1ff0-11e5-8f9d-94ac51681c65.png)

Code to reproduce:

``` php
public function behaviors()
{
    return [
        'access' => [
            'class' => AccessControl::className(),
            'rules' => [
                [
                    'allow' => true,
                    'roles' => ['@'],
                    'matchCallback' => function ($rule, $action) {
                        echo 'Yii::$app->controller->actionParams = '.print_r(Yii::$app->controller->actionParams, TRUE);
                        echo '$action->controller->actionParams = '.print_r($action->controller->actionParams, TRUE);
                        echo '$action = '.print_r($action, TRUE);
                        die();
                    }
                ],
            ],
        ]
    ];
}
```

Is this a bug or by design (or maybe I made a mistake and this shouldn't happen)?

If by design, then, please, provide any explanation or reason, for which `actionParams` can't be accessed / are empty inside `matchCallback`?

Using `$_GET` inside `matchCallback` is possible and is some-kind-of workaround, but I consider this solution as not professional one and would like to know, how can I use `actionParams` inside `matchCallback` in ACF or why can't I use it there?

Additional reading:
- [Stack Overflow question](http://stackoverflow.com/q/31158386/1469208),
- [Yii forum post](http://www.yiiframework.com/forum/index.php/topic/64662-reading-action-params-in-acfs-matchcallback)

**Comment by lynicidn** [comment]
https://github.com/yiisoft/yii2/issues/5886#issuecomment-61487679

**Comment by lynicidn** [comment]
https://github.com/yiisoft/yii2/issues/1796#issuecomment-31625194

**Comment by trejder** [comment]
This seems to be by design -- ACF runs at much earlier stage than params binding [source](http://stackoverflow.com/a/31159488/1469208):

> Controller class has method `bindActionParams()` which is invoked by *\yii\base\Action* when it begins to run with the given parameters. So at first controller invoke an action method and run it with params and then action bind its params to controller. But *AccessControl* check is executing much earlier before action is running, that's why `actionParams` are empty.

Another workaround (same source):

> (...) consider use of `Yii::$app->request->get()` which also will test parameter for *isset*.

For the reasons explained in issue (requiring action params to evaluate, if user should be allowed to execute particular action) it would be great to have `actionParams` available inside ACF's `matchCallback`. So, consider this as a feature request. However, I understand, that in above mentioned circumstances implementing this may be problematic, if not impossible. Issue can be closed. Thank you.

**Comment by samdark** [comment]
Won't be implemented in 2.0.
